From ef5c40e46da20b5b72d6e42e0a4f1fe7a83cc417 Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 24 Sep 2014 17:28:04 -0400
Subject: [net] svcrdma: remove rdma_create_qp() failure recovery logic

Message-id: <68f42052863248d12a7aa80b701de51c4edaff35.1411578494.git.dledford@redhat.com>
Patchwork-id: 96679
O-Subject: [Patch RHEL7 11/26] svcrdma: remove rdma_create_qp() failure recovery logic
Bugzilla: 1107621
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Steve Wise <swise@opengridcomputing.com>

In svc_rdma_accept(), if rdma_create_qp() fails, there is useless
logic to try and call rdma_create_qp() again with reduced sge depths.
The assumption, I guess, was that perhaps the initial sge depths
chosen were too big.  However they initial depths are selected based
on the rdma device attribute max_sge returned from ib_query_device().
If rdma_create_qp() fails, it would not be because the max_send_sge and
max_recv_sge values passed in exceed the device's max.  So just remove
this code.

Signed-off-by: Steve Wise <swise@opengridcomputing.com>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
(cherry picked from commit d1e458fe671baf1e60afafc88bda090202a412f1)
Signed-off-by: Doug Ledford <dledford@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/sunrpc/xprtrdma/svc_rdma_transport.c b/net/sunrpc/xprtrdma/svc_rdma_transport.c
index 06a5d92..374feb4 100644
--- a/net/sunrpc/xprtrdma/svc_rdma_transport.c
+++ b/net/sunrpc/xprtrdma/svc_rdma_transport.c
@@ -943,23 +943,8 @@ static struct svc_xprt *svc_rdma_accept(struct svc_xprt *xprt)
 
  ret = rdma_create_qp(newxprt->sc_cm_id, newxprt->sc_pd, &qp_attr);
  if (ret) {
-  /*
-   * XXX: This is a hack. We need a xx_request_qp interface
-   * that will adjust the qp_attr's with a best-effort
-   * number
-   */
-  qp_attr.cap.max_send_sge -= 2;
-  qp_attr.cap.max_recv_sge -= 2;
-  ret = rdma_create_qp(newxprt->sc_cm_id, newxprt->sc_pd,
-         &qp_attr);
-  if (ret) {
-   dprintk("svcrdma: failed to create QP, ret=%d\n", ret);
-   goto errout;
-  }
-  newxprt->sc_max_sge = qp_attr.cap.max_send_sge;
-  newxprt->sc_max_sge = qp_attr.cap.max_recv_sge;
-  newxprt->sc_sq_depth = qp_attr.cap.max_send_wr;
-  newxprt->sc_max_requests = qp_attr.cap.max_recv_wr;
+  dprintk("svcrdma: failed to create QP, ret=%d\n", ret);
+  goto errout;
  }
  newxprt->sc_qp = newxprt->sc_cm_id->qp;
 
-- 
1.7.1